mission/mission_handler_landing.hxx

```diff
@@ -0,0 +1,193 @@
+{
+    bool success = true;
+
+    print_util_dbg_print("Automatic-landing: descent_to_small_altitude\r\n");
+    auto_landing_behavior_ = DESCENT_TO_SMALL_ALTITUDE;
+    state_.mav_mode_custom &= static_cast<Mav_mode::custom_mode_t>(0xFFFFFFE0);
+    state_.mav_mode_custom |= Mav_mode::CUST_DESCENT_TO_SMALL_ALTITUDE;
+    is_landed_ = false;
+
+    waypoint_ = wpt;
+
+    return success;
+}
+
+template <class T1, class T2>
+Mission_handler::update_status_t Mission_handler_landing<T1, T2>::update()
+{
+    /*****************************
+    Handle internal landing states 
+    *****************************/
+    // Determine if we should switch between the landing states
+    bool next_state = false;
+    if (auto_landing_behavior_ == DESCENT_TO_GND)
+    {
+        alt_lpf_ = LPF_gain_ * (alt_lpf_) + (1.0f - LPF_gain_) * ins_.position_lf()[Z];
+        if ((ins_.position_lf()[Z] > -0.1f) && (maths_f_abs(ins_.position_lf()[Z] - alt_lpf_) <= 0.2f))
+        {
+            next_state = true;
+        }
+    }
+    else if (auto_landing_behavior_ == DESCENT_TO_SMALL_ALTITUDE)
+    {
+        float z_lim = desc_to_ground_altitude_- desc_to_ground_range_;
+        if ((ins_.position_lf()[Z]) > z_lim)                                         // Drone is below desc2gnd waypoint (for cases when we start landing below it)
+        {
+            next_state = true;
+        }
+    }
+    
+    /*************
+    Set controller 
+    *************/
+    bool ret = false;
+    switch (auto_landing_behavior_)
+    {
+        case DESCENT_TO_SMALL_ALTITUDE:
+            ret = set_desc_to_small_alt_control_command();
+            break;
+
+        case DESCENT_TO_GND:
+            ret = set_desc_to_ground_control_command();
+            break;
+    }
+
+    // If we are switching between states, ... then switch
+    if (next_state)
+    {
+        switch (auto_landing_behavior_)
+        {
+            case DESCENT_TO_SMALL_ALTITUDE:
+                print_util_dbg_print("Automatic-landing: descent_to_GND\r\n");
+                auto_landing_behavior_ = DESCENT_TO_GND;
+                state_.mav_mode_custom &= static_cast<Mav_mode::custom_mode_t>(0xFFFFFFE0);
+                state_.mav_mode_custom |= Mav_mode::CUST_DESCENT_TO_GND;
+                alt_lpf_ = ins_.position_lf()[Z];
+                break;
+
+            case DESCENT_TO_GND:
+                print_util_dbg_print("Auto-landing: disarming motors \r\n");
+                auto_landing_behavior_ = DESCENT_TO_SMALL_ALTITUDE;
+                //Do not reset custom flag here, to be able to check after landing
+                // in case something went wrong. Is reset while arming
+                state_.set_armed(false);
+                state_.mav_state_ = MAV_STATE_STANDBY;
+                is_landed_ = true;
+                break;
+        }
+    }
+    
+
+    /********************
+    Determine status code 
+    ********************/
+    if (waypoint_.autocontinue() == 1 && is_landed_)
```

Comment:
Sounds like copy paste error. I would even think the mission is finished if we landed and the waypoint is **not** autocontinue